### PR TITLE
Fix notifications [WIP]

### LINKED ISSUE #679: Reset notification cache
Need to reset notification cache when accepting/declining connection or pack requests

### LINKED ISSUE #680: Display pack requests in notification
Need to properly display pack requests. Also need to update to "GroupJoinRequest" from "GroupJoinRequests" in serialization when Josh deploys new version

### PR DESCRIPTION
**Pull request by sunyatax**
- Fix #679 

Waiting on API to return "creator" instead of "user" in group response data to Fix #680

### CODE REVIEW

File: lib/screens/notifications/widgets/connection_request_response.dart [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+  void _respondToRequest(BuildContext context, bool response) async {
+    // show Junto loader
+    JuntoLoader.showLoader(context);
+    // respond to request
+    await Provider.of<UserRepo>(context, listen: false)
+        .respondToConnection(userAddress, response);
+    // delete notification from cache
+    await Provider.of<LocalCache>(context, listen: false)
+        .deleteNotification(notification.address);
+    // refetch notifications
```
Comment (orestesgaolin) [inline lib/screens/notifications/widgets/connection_request_response.dart:25]: Don't access notifications through the `LocalCache`. Do it through `NotificationsHandler` as `NotificationsHandler` stores a list of current notifications in memory. Removing it from cache won't remove it from UI. 

Also, don't access repositories/cache in widgets directly.

File: lib/backend/services/hive_service.dart [hunk 2]
```diff
@@ -78,21 +79,33 @@ class HiveCache implements LocalCache {
       final box = await Hive.openLazyBox<JuntoNotification>(
           _supportedBox[DBBoxes.notifications]);
       List<JuntoNotification> items = [];
-      for (dynamic key in box.keys) {
+      for (String key in box.keys) {
+        key.toString();
```
[the remark this answers is not in the document]
Comment (sunyatax) [reply lib/backend/services/hive_service.dart:83]: Woops, didn't mean to commit this... was trying to debug the type issue we were getting on retrieveNotifications()

File: lib/screens/notifications/widgets/connection_request_response.dart (hunk 1 above)
Comment (sunyatax) [reply lib/screens/notifications/widgets/connection_request_response.dart:25]: Deleting it via LocalCache produces the desired result. I tried it via NotificationsHandler and it doesn't get removed from the UI. The issue is that the notifications are still being persisted in the Hive cache, despite being deleted from the API is it not? If so, can you clarify why the LocalCache approach isn't correct?

Comment (orestesgaolin) [reply lib/screens/notifications/widgets/connection_request_response.dart:25]: Don't access LocalCache directly in widgets, but expose operations on notifications via NotificationsHandler. Of course the final result should be to remove notifications from cache, but do it through NotificationsHandler similarly to how notifications are marked as read. This way we won't have to touch widgets code if we want to modify how are we caching notifications.

**Pull request merged**